readme.md

```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    type MemberPositions = 'intern' | 'developer' | 'tech-lead';
+
+    interface Person {
+        firstName: string;
+        lastName: string;
+        yearsOfExperience: number;
+    }
+
+    const team: Record<MemberPositions, Person[]> = {
+        intern: [],
+        developer: [],
+        ['tech-lead']: [],
+    };
+
+    // Our team has hired a new member.
+    team.intern.push({ firstName: 'John', lastName: 'Doe', yearsOfExperience: 0 });
+
+    // Record forces you to initialize all of the property keys
+    // TypeScript Error: "tech-lead" property is missing
+    const teamEmpty: Record<MemberPositions, null> = {
+        intern: null,
+        developer: null,
+    };
+    ```
+    </details>
+
 - [`Exclude<T, U>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1436-L1439) - Exclude from `T` those types that are assignable to `U`.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/JYOwLgpgTgZghgYwgAgMIHsQ2Ac2QbwChlkBbCMAC3QBMAuZAZzClBwG5jk5GBPEBAwBG6dABsIcEJxIBXKGIbNWIDlwCOs6LyUs2nAL6FCAegBUZ5AHEKyWSGCZk6GMgDyQgFYQEYACq8AA4QAOTIANYQvIzIVHBgyBAAHpAgNMh+cFA4FAHByGYmhGBBKADSUYxuMHkQADwe3r61ADQZWTn+pQB8yAC8BFwA2hW8yKARUS7uXj5dwQC6DI1ztSNRC4kpEGkxmdm5pcgA-MijyAwgEABu0IQGQ5G80yvNpQucphbupMAJVOhGChAlB0MEoGBgBAYnEEslUukAKJJBBiWQ0CC1ApFEr5Ny-eYQUaMBqzN7BNrI1HozE9frIAAKwAQ4TqXFehJaXCpaIxdSeLzJnLOlWqtVJTWFPJptW63UI3U+uJQAGU9KoMFhcMT6cSxaU6prsDg2so2IrkCYTMgQuQATQwgAfG3yMROm2abQhYzmMzESwAMVBpEmz1cRtwDAABraKNQHchnSEePwEO6Qq7056oLwQlH-VtqRiC3qagaIyamOqcN1o7H7enM4mbSmBHmCxI4Nc2MgY620-nscUjgA5TBqlQ4Cs6gbSvkC8OYY1tUviium6tyz6+6y2TBiMYgsHQSHQ2KUeLIGjoZAgdBw7ZpWJHM2qZAwYPP-IVofK5BjkAJzYH8BnxP5amJQ0l1wDdJ0VYwEEwZhb3HasGAAoCNWgvABnwbg+AEBgWC0ZADE4IA">
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    interface Config {
+        method: string;
+        async: boolean;
+        url: string;
+        query: string;
+    }
+
+    /** Get union of ObjectType' keys that extend TargetType */
+    type KeysOfType<ObjectType, TargetType> = {
+        [Key in keyof ObjectType]: ObjectType[Key] extends TargetType ? Key : never;
+    }[keyof ObjectType];
+
+
+    /** Omit those properties that extend ExcludeType */
+    type OmitTypeKeys<ObjectType, ExcludeType> = Pick<
+        ObjectType,
+        Exclude<keyof ObjectType, KeysOfType<ObjectType, ExcludeType>>
+    >;
```

Comment:
I don't think the example should require helper types. This example is more complicated than it needs to be.